### PR TITLE
Fix navbar's active item highlight

### PR DESCRIPTION
**Pull request by rapzo**

### CODE REVIEW

**Comment by rapzo** [comment]
Not my best ruby-fu applied here. Feel free to throw your tips on how to ruby ;-)

**Comment by tsubik** [comment]
Thanks. I will take a look.

**Comment by tsubik** [comment]
Just wondering about this "about page" :smile: 

There is `about` action in home controller, but about page is a dynamic one. I think it's kinda misleading. I will try to remove that empty page and see if the menu works after that.

**Comment by tsubik** [comment]
Actually on prod I see there is no about page. That's just the menu header name.

**Comment by rapzo** [comment]
Yes, it was my doings (the about action in the home controller and all)...
But, even if it doesn't exist, i thing the helper will always return false for its path analysis and go about the paths from the dynamic generated. It should be safe either to keep or remove it IMO.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -16,4 +16,22 @@ def render_breadcrumb
...
   end
+
+  def active_menu_page?(**args)
+    unless args.nil?
+      has_pages = false
+      has_path = false
+
+      has_pages = args[:pages].any? { |page| current_page?(page[:slug]) } unless args[:pages].nil?
+
+      has_path = current_page?(args[:path]) unless args[:path].nil?
```
Comment (aabdaab) [inline app/helpers/application_helper.rb]: It's not a blocker but a matter of convention comment; `hasSomething` convention in javascript becomes `something?` in ruby - `?` in ruby indicates that a method returns a boolean value (like internal blank?, valid? etc).

Comment (tsubik) [reply app/helpers/application_helper.rb]: I'm actually refactoring. So all that will change.

Comment (aabdaab) [reply app/helpers/application_helper.rb]: 👍

Comment (tsubik) [reply app/helpers/application_helper.rb]: Trying to make sectors and companies controller actions to also highlight the TPI menu

Comment (rapzo) [reply app/helpers/application_helper.rb]: Oh... Didn't go over those... Sorry.

File: app/javascript/components/DropdownSelector.js [hunk 2]
```diff
@@ -171,40 +189,62 @@ const DropdownSelector = ({ sectors, companies, selectedOption, defaultFilter =
             <div className="dropdown-selector__options-wrapper">
               {isFilterBySector && (
                 <div className="dropdown-selector__options">
-                  {options.map(option => (
-                    <div onClick={() => handleOptionClick(option)} className="dropdown-selector__option">
+                  {(options.length > 0 && options.map((option, i) => (
```
Comment (aabdaab) [inline app/javascript/components/DropdownSelector.js]: no need to check if it's `> 0` - `options.length` will evaluate to `false` if it's null or 0

File: app/javascript/components/DropdownSelector.js [hunk 3]
```diff
@@ -171,40 +189,62 @@ const DropdownSelector = ({ sectors, companies, selectedOption, defaultFilter =
...
+                    <div
+                      onClick={() => handleOptionClick(option)}
+                      className="dropdown-selector__option"
+                      key={`${option.name}-${i}`}
+                    >
                       {option.name}
                     </div>
+                  ))) || (searchValue.length > 0 && options.length === 0 && (
```
Comment (aabdaab) [inline app/javascript/components/DropdownSelector.js]: same here: you can remove `> 0` bit and instead of explicitly checking if `options.length === 0` turn it into `!options.length`

File: app/javascript/components/DropdownSelector.js [hunk 4]
```diff
@@ -171,40 +189,62 @@ const DropdownSelector = ({ sectors, companies, selectedOption, defaultFilter =
...
                   ))}
-                  {searchValue && !options.length && <div>No results found.</div>}
                 </div>
               )}
               {isFilterByCompany && (
                 <div className="dropdown-selector__options">
-                  {Object.keys(companiesBySector).map(sector => (
-                    <>
-                      <div className="dropdown-selector__sector-name">
+                  {(options.length > 0 && Object.keys(companiesBySector).map((sector, i) => (
```
Comment (aabdaab) [inline app/javascript/components/DropdownSelector.js]: and here

File: app/javascript/components/DropdownSelector.js [hunk 5]
```diff
@@ -171,40 +189,62 @@ const DropdownSelector = ({ sectors, companies, selectedOption, defaultFilter =
...
+                          key={`${option.name}-${i}-${j}`}
                         >
                           {option.name}
                         </div>
                       ))}
-                    </>
-                  ))}
-                  {searchValue && !options.length && <div>No results found.</div>}
+                    </Fragment>
+                  ))) || (searchValue.length > 0 && options.length === 0 && (
```
Comment (aabdaab) [inline app/javascript/components/DropdownSelector.js]: I see that this is repeating, maybe move it to a variable

**Comment by tsubik** [comment]
Ok, done some little ruby refactoring, hopefully didn't break anything.

File: app/javascript/components/DropdownSelector.js (hunk 2 above)
Comment (rapzo) [reply app/javascript/components/DropdownSelector.js]: i know what you mean but readability speaking I only want something to happen when that condition is true. If options was an object with a map function and a length property, say, a string, the sole condition of checking the property would pass whereas the condition for checking its length won't.
TL;DR it's like `i++` vs `i += 1` readability vs habits

File: app/javascript/components/DropdownSelector.js (hunk 3 above)
Comment (rapzo) [reply app/javascript/components/DropdownSelector.js]: Same motif as in the comment before.

**Comment by rapzo** [comment]
Updated. Tested it and all seems good to go!
